bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -317,29 +214,22 @@ public void init() {
 
                 // OT-34: disable create group menu if no guide stars defined
                 _w.newMenu.addMenuListener(new MenuListener() {
-                    @Override
-                    public void menuSelected(MenuEvent e) {
-                        guideGroupMenu.setEnabled(obsComp.getTargetEnvironment().getGuideEnvironment().getTargets().size() != 0);
-                    }
-
-                    @Override
-                    public void menuDeselected(MenuEvent e) {
-                    }
-
-                    @Override
-                    public void menuCanceled(MenuEvent e) {
+                    @Override public void menuSelected(final MenuEvent e) {
+                        guideGroupMenu.setEnabled(obsComp.getTargetEnvironment().getGuideEnvironment().getTargets().nonEmpty());
                     }
+                    @Override public void menuDeselected(final MenuEvent e) {}
+                    @Override public void menuCanceled(final MenuEvent e) {}
                 });
             }
         }
-        _w.positionTable.reinit(getDataObject());
+        _w.positionTable.reinit(obsComp);
         _w.guidingControls.manualGuideStarButton().peer().setVisible(GuideStarSupport.supportsManualGuideStarSelection(getNode()));
         updateGuiding();
         _agsPub.watch(getContextObservation());
     }
 
     // OtItemEditor
-    protected void cleanup() {
+    @Override protected void cleanup() {
         _agsPub.watch(null);
```

Comment:
Looking at `AgsContextPublisher`, this seems to be a really horrible trick that basically says to:
1. Unwatch any currently watched observations; and
2. Clear out the `AgsContext` by setting it to null through a bit of ridiculousness using `ImOption` via a call to an `AgsContext.create` method with `null`, instead of simply doing a `null` check and if null, setting the `AgsContext` to `null` directly.

Seems completely straightforward, no? :laughing: 
This should probably be simplified to not be so bizarre, perhaps by the addition of an `unwatch` method or probably better by adding a `watch(Option[ISPObservation])` method, as calling with `None` seems like it makes a lot more sense and wouldn't introduce any issues with regards to the `AgsContext`.

There is also an unused method in `AgsContextPublisher`. I think I will look at simplifying this, and thanks for pointing it out.